src/App.test.jsx

```diff
@@ -1,16 +1,22 @@
 import React from 'react';
 
-import { render } from '@testing-library/react';
+import { render, fireEvent } from '@testing-library/react';
 
 import App from './App';
 
 test('App', () => {
-  const { getByText } = render((
+  const { getByText, getByLabelText } = render((
     <App />
   ));
 
   expect(getByText(/추가/)).not.toBeNull();
+  const value = '새로운 할 일';
 
-  // TODO: 통합 테스트 코드 작성
-  // CodeceptJS => 실제 브라우저에서 사용자 테스트 실행 가능.
+  fireEvent.change(getByLabelText(/할 일/), { target: { value } });
+  fireEvent.click(getByText(/추가/));
+
+  expect(getByText(value)).not.toBeNull();
+
+  expect(getByText(/완료/)).not.toBeNull();
+  fireEvent.click(getByText(/완료/));
```

Comment:
```suggestion
  fireEvent.change(getByLabelText(/할 일/), { target: { value } });
  fireEvent.click(getByText(/추가/));

  expect(getByText(value)).not.toBeNull();

  fireEvent.click(getByText(/완료/));
  
  expect(getByText(/완료/)).toBeNull();
```

완료 동작의 의미를 좀 더 드러내봤습니다.

Reply:
피드백을 보기 직전에 한 새로운 커밋에서는 ```App.test.jsx```를 많이 손봤습니다. 

redux를 사용해서 상태관리를 하도록 바꾸니까, 기존의 ```App.test.jsx```에서 에러가 나더라고요! 그래서, 똑같이 ```useDispatch```, ```useSelector```를 써서 테스트도 바꿔줬습니다. 

1) 바뀐 테스트에서도 완료동작의 의미를 조금더 드러내는 것이 좋을까요?
2) 또, App 테스트에서도 redux 구조를 사용해서 테스트를 하는게 맞는지 피드백 해주시면 좋겠습니다. 제가 코드를 잘못짜서(?) 기존 앱 테스트가 동작을 잘 안 하는건지, App에서도 (바뀐 ```App.test.jsx```처럼) redux 구조를 사용해서 테스트를 해야하는건지 궁금합니다.

Reply:
1. 이정도면 '큰 문제 없이 작동을 보장할 수 있다' 는 본인이 정하면 됩니다.

2. 강의를 보시면 아시겠지만 App.test.jsx도 당연히 변경되어야 합니다.

useSelector에 대한 mockImplementation을 하는 걸 보실 수 있을 겁니다.

자식 중에 useSelector를 사용하는 애들이 있기 때문이죠ㅎㅎ

Reply:
그렇군요 감사합니다!! 👍